### ISSUE TITLE
CLAVRx Geotiff file creation failing on AVHRR  sensors

### ISSUE THREAD

**Issue by kathys** [ssec/polar2grid#517]
When testing and comparing the generation of Geotiff files from input CLAVRx hdf files between P2G 2.3 and v3.0, I found that V3.0 did not create any output images from input AVHRR data sets. This consisted of Metop-B, -C and NOAA-18,-19 instruments.  I was trying to create just the default products.  The error message in all of the log files says 

File "/data/users/kathys/polar2grid-swbundle-20220718-201315/libexec/python_runtime/lib/python3.10/site-packages/pyresample/spherical.py", line 233, in angle
    raise ValueError("No common point in angle computation.")
ValueError: No common point in angle computation.

Example log file can be found here: `/data/users/kathys/test_data/clavrx/noaa19_v3`

The software did produce output for the VIIRS and MODIS sensors.

**Comment by kathys** [comment]
I have not tested other reader/writers yet, such as AWIPS.

**Comment by djhoese** [comment]
For my own future reference here is the full error:

```
  File "/data/users/kathys/polar2grid-swbundle-20220718-201315/libexec/python_runtime/lib/python3.10/site-packages/polar2grid/filters/_filter_scene.py", line 93, in filter_scene
    input_scene = _filter_scene_day_only_products(
  File "/data/users/kathys/polar2grid-swbundle-20220718-201315/libexec/python_runtime/lib/python3.10/site-packages/polar2grid/filters/_filter_scene.py", line 70, in _filter_scene_day_only_products
    return day_filter.filter_scene(input_scene)
  File "/data/users/kathys/polar2grid-swbundle-20220718-201315/libexec/python_runtime/lib/python3.10/site-packages/polar2grid/filters/_base.py", line 86, in filter_scene
    if not self._filter_data_array(scene[data_id], _cache):
  File "/data/users/kathys/polar2grid-swbundle-20220718-201315/libexec/python_runtime/lib/python3.10/site-packages/polar2grid/filters/day_night.py", line 137, in _filter_data_array
    slc = _get_sunlight_coverage(area, mid_time, sza_threshold)
  File "/data/users/kathys/polar2grid-swbundle-20220718-201315/libexec/python_runtime/lib/python3.10/site-packages/polar2grid/filters/day_night.py", line 50, in _get_sunlight_coverage
    adp = polygon_for_area(area_def)
  File "/data/users/kathys/polar2grid-swbundle-20220718-201315/libexec/python_runtime/lib/python3.10/site-packages/polar2grid/filters/_utils.py", line 86, in polygon_for_area
    boundary = boundary_for_area(area_def)
  File "/data/users/kathys/polar2grid-swbundle-20220718-201315/libexec/python_runtime/lib/python3.10/site-packages/polar2grid/filters/_utils.py", line 80, in boundary_for_area
    adp = AreaDefBoundary(area_def, frequency=int(area_def.shape[0] * freq_fraction))
  File "/data/users/kathys/polar2grid-swbundle-20220718-201315/libexec/python_runtime/lib/python3.10/site-packages/pyresample/boundary.py", line 91, in __init__
    lons, lats = area.get_bbox_lonlats()
  File "/data/users/kathys/polar2grid-swbundle-20220718-201315/libexec/python_runtime/lib/python3.10/site-packages/pyresample/geometry.py", line 310, in get_bbox_lonlats
    if force_clockwise and not self._corner_is_clockwise(
  File "/data/users/kathys/polar2grid-swbundle-20220718-201315/libexec/python_runtime/lib/python3.10/site-packages/pyresample/geometry.py", line 376, in _corner_is_clockwise
    angle = arc1.angle(arc2)
  File "/data/users/kathys/polar2grid-swbundle-20220718-201315/libexec/python_runtime/lib/python3.10/site-packages/pyresample/spherical.py", line 233, in angle
    raise ValueError("No common point in angle computation.")
ValueError: No common point in angle computation.
```

This is during the portion of the code that filters out day-only products when it is night time data (I think). It seems to be failing when it gets to `cld_opd_dcomp`. Is that a day-only product (the "d" in "dcomp")? I'll have to check but my best guess is the lon/lats have a lot of NaNs in them.

**Comment by kathys** [comment]
Yes, it is a product that is only created during the day.

**Comment by djhoese** [comment]
I've started diving into this. It looks like the entire top 9 rows of the lon/lat arrays are all NaNs. I'll see if I can improve the NaN handling in the error'ing parts of the code.

**Comment by djhoese** [comment]
@kathys I'm not sure how we should handle this. The way this works right now is that it takes the lons and lats as two 2D arrays and makes a "polygon" of the outer most rows and columns. In this case the coordinates for the "top" of this polygon are all NaNs and so we fail. I could try to look at more data, for example the outer dask "chunk" of the array and determine the bounds for that chunk, but there is still a chance of data cases where the entire chunk is NaNs. So we still need a procedure for how to handle those cases. The main question that will lead us to an answer is how often/common are these top NaN values in AVHRR or CLAVRx output?

Possible solutions:

1. If polygon creation fails (there are NaNs), log a warning and don't do the filtering.
2. Same as 1, but fallback to loading entire lon/lat arrays and compute min/max bounding box and use that.
3. Rewrite polygon creation to progressively search for more valid data (I don't see a benefit of this over 2).

**Comment by djhoese** [comment]
Forgot to add, these polygons are used for day filtering, night filtering, and grid coverage filtering.

**Comment by kathys** [comment]
I just did a quick test on samples from all 4 AVHRR instruments that we support, Metop-B, Metop-C, NOAA-18 and NOAA-19.  I chose recent data.  I was successfully able to create AVHRR L1b images from all 4 instruments with no problems, but was not successfully able to create CLAVRx images from any of them. Given this, it seems to be a problem in CLAVRx and not in the AVHRR data, and looks like it always happens.  

I think the best option that you outline is option `2`.  I am not sure that it is necessary for AVHRR L1B, but might be a good fallback I guess.

**Comment by djhoese** [comment]
Can you point me to some of the AVHRR L1b files? I'll look at them in Satpy just to make sure. They may not being going through the same filtering that the clavrx products are.

**Comment by kathys** [comment]
`bumi:/data/users/kathys/test_data/avhrr/2022_examples`

**Comment by djhoese** [comment]
Yep, looked at NOAA-19. Definitely no NaNs in the lon/lats for band 1, but NaNs in the `cld_opd_dcomp` product lons/lats.
